src/lib/Ska/Starcheck/FigureOfMerit.pm

```diff
@@ -28,42 +24,66 @@ our @EXPORT = qw();
 our @EXPORT_OK = qw( make_figure_of_merit );
 our %EXPORT_TAGS = ( all => \@EXPORT_OK );
 
+use Inline Python => q{
+import sys
+import perl
+# this provides an interface back to the Perl namespace
+
+try:
+    from starcheck.star_probs import acq_success_prob, prob_n_acq, mag_for_p_acq
+except ImportError as err:
+    # write errors to starcheck's global warnings and STDERR
+    perl.warning("Error with Inline::Python imports \n".format(err))
+
+def _acq_success_prob(date, t_ccd, mag, color, spoiler):
+    out = acq_success_prob(date, t_ccd, mag, color, spoiler)
+    return out.tolist()
 
-our $CUM_PROB_LIMIT = -5.2;
+def _prob_n_acq(acq_probs):
+    n_acq_probs, n_or_fewer_probs = prob_n_acq(acq_probs)
+    return n_acq_probs.tolist(), n_or_fewer_probs.tolist()
+};
 
+our $CUM_PROB_LIMIT = 10 ** -5.2;
 
 sub make_figure_of_merit{
     my $c;
     my $self = shift;
     return unless ($c = $self->find_command("MP_STARCAT"));
+
     my @probs;
     my %slot_probs;
+
+    my $t_ccd = 19.0;  # ??
+    my $date = $c->{date};
+
     foreach my $i (1..16) {
-	my %acq = ();
-	my $type = $c->{"TYPE$i"};
-	if ($type =~ /BOT|ACQ/) {
-	    $acq{index} = $i;
-	    $acq{magnitude} = $c->{"GS_MAG$i"};
-	    $acq{warnings} = [grep {/\[\s{0,1}$i\]/} (@{$self->{warn}}, @{$self->{yellow_warn}})];
-            $acq{n100_warm_frac} = $self->{n100_warm_frac} || $self->{config}{n100_warm_frac_default};
-	    #find the probability of acquiring a star
-            my $star_prob = star_prob(\%acq);
+	if ($c->{"TYPE$i"} =~ /BOT|ACQ/) {
+	    my $mag = $c->{"GS_MAG$i"};
+            my @warnings = grep {/\[\s{0,1}$i\]/} (@{$self->{warn}}, @{$self->{yellow_warn}});
+            my $spoiler = grep(/Search Spoiler/, @warnings) ? 1 : 0;
+            my $color = $c->{"GS_BV$i"};
+            my $star_prob = _acq_success_prob($date, $t_ccd, $mag, $color, $spoiler);
 	    push @probs, $star_prob;
             $slot_probs{$c->{"IMNUM$i"}} = $star_prob;
+            $c->{"P_SUCC$i"} = $star_prob;
 	}
     }
-#    push @{$self->{yellow_warn}}, "Probs = " .  join(" ",  map{ sprintf("%.4f",$_) } @probs) . "\n";
-
-    #calculate the probability of acquiring n stars
-    my $n_slot = $#probs+1;
-    my @nacq_probs = prob_nstars($n_slot, @probs);
-    #calculate the probability of acquiring at least n stars
-    $self->{figure_of_merit} = cum_prob($n_slot, @nacq_probs);
     $self->{acq_probs} = \%slot_probs;
 
+    # Calculate the probability of acquiring n stars
+    my ($n_acq_probs, $n_or_fewer_probs) = _prob_n_acq(\@probs);
+
+    $self->{figure_of_merit} = {expected => substr(sum(@probs), 0, 4),
+                                cum_prob => [map { log($_) / log(10.0) } @{$n_or_fewer_probs}],
+                                cum_prob_bad => ($n_or_fewer_probs->[2] > $CUM_PROB_LIMIT)
+                                };
 
+    # Dynamic mag limits based on 75% and 50% chance of successful star acq
+    $self->{mag_faint_yellow} = mag_for_p_acq(0.75, $date, $t_ccd);
```

Comment:
That makes sense.  I was originally thinking about only applying the dynamic limits after a time T to avoid the non-informative regress diffs but the functional change right here would be as effective and less code messy.